opensrp-reveal/src/main/java/org/smartregister/reveal/sync/RevealClientProcessor.java

```diff
@@ -165,28 +149,55 @@ private String processSprayEvent(Event event, ClientClassification clientClassif
     private String processMosquitoCollectionEvent(Event event, ClientClassification clientClassification, boolean localEvents) {
         String operationalAreaId = null;
         if (event.getDetails() != null && event.getDetails().get(TASK_IDENTIFIER) != null) {
-            // todo: add logic to update business status when data dictionary is available
+            operationalAreaId = updateTask(event, localEvents);
+
             try {
-                // todo: add status when data dictionary specifies how to collect and determine its value
-                String startDate = event.findObs(null, false, JsonForm.TRAP_SET_DATE).getValue().toString();
-                String endDate = event.findObs(null, false, JsonForm.TRAP_FOLLOW_UP_DATE).getValue().toString();
-                event.getDetails().put(START_DATE, startDate);
-                event.getDetails().put(END_DATE, endDate);
+                Obs startDateObs = event.findObs(null, false, JsonForm.TRAP_SET_DATE);
+                Obs endDateObs = event.findObs(null, false, JsonForm.TRAP_FOLLOW_UP_DATE);
+
+                Object startDate = startDateObs == null ? null : startDateObs.getValue();
+                Object endDate = endDateObs == null ? null : endDateObs.getValue();
+
+                event.getDetails().put(START_DATE, startDate == null ? null : startDate.toString());
+                event.getDetails().put(END_DATE, endDate == null ? null : endDate.toString());
+
+                RevealClientProcessor clientProcessor = getInstance(RevealApplication.getInstance().getApplicationContext());
+                event.getDetails().put(STATUS, clientProcessor.calculateBusinessStatus(event));
```

Comment:
Remove this block 155-165. Implement in `ec_client_fields.json`

e.g

```
 {
          "column_name": "status",
          "type": "Event",
          "json_mapping": {
                  "field": "obs.fieldCode",
                  "concept": "business_status"
          }
        }
 
```

Reply:
Done.